tests/ha/control_pod/test_control_pod_restart.py

```diff
@@ -747,161 +747,97 @@ def test_mpu_after_ctrl_pod_failover(self):
-        resp = self.ha_obj.failover_pod(pod_obj=self.node_master_list[0], pod_yaml=pod_yaml,
-                                        failover_node=failover_node)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 2: Successfully failed over control pod to %s. Cluster is in good state",
-                    failover_node)
-
-        for cnt in range(HA_CFG["common_params"]["loop_count"]):
-            LOGGER.info("Loop: %s", cnt)
-            control_pods = self.node_master_list[0].get_pods_node_fqdn(
-                const.CONTROL_POD_NAME_PREFIX)
-            control_pod_name = list(control_pods.keys())[0]
-            control_node = control_pods.get(control_pod_name)
-            LOGGER.info("Control pod %s is hosted on %s node", control_pod_name, control_node)
-
-            failover_node = self.system_random.choice([ele for ele in self.host_worker_list if
-                                                       ele != control_node])
-            LOGGER.debug("Fail over node is: %s", failover_node)
-
-            LOGGER.info("Step 3: Failover control pod %s to node %s by changing its NodeSelector "
-                        "and check cluster status. \nFailover count: %s", control_pod_name,
-                        failover_node, cnt)
-            resp = self.ha_obj.change_pod_node(self.node_master_list[0],
-                                               pod_node={control_pod_name: failover_node})
-            assert_utils.assert_true(resp[0], resp)
-            LOGGER.info("Check cluster status")
+        for count in range(num_users):
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[count],
+                                                        log_prefix=self.test_prefix,
+                                                        skipcleanup=True)
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects for %s IAM "
+                    "users created.", num_users)
+        LOGGER.info("Step 2: Shutdown %s control pods in loop", self.repl_num - 1)
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        pod_left = None
+        for loop in range(len(pod_list)):
+            LOGGER.info("Shutting down %s control pods for loop: %s", self.repl_num - 1, loop)
+            delete_pods = list()
+            while pod_left not in delete_pods:
+                delete_pods.extend(self.system_random.sample(pod_list, self.repl_num - 1))
+            for pod in delete_pods:
+                resp = self.node_master_list[0].create_pod_replicas(num_replica=0, pod_name=pod)
+                assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3: Check cluster status")
             resp = self.ha_obj.poll_cluster_status(self.node_master_list[0])
```

Comment:
Should we give some timeout here?
Or should we change default timeout of this function?

Reply:
I guess we already have enough sleep in pod replicas methods and also check if replicas are scaled. So still we want to add timeout here? Or u think the cluster shud be up early than 20mins?